trieste/acquisition/rule.py

```diff
@@ -211,31 +212,242 @@ def acquire(
         return points
 
 
-class AsynchronousGreedy(
-    AcquisitionRule[State[Optional["AsynchronousGreedy.State"], TensorType], SP_contra]
+@dataclass(frozen=True)
+class AsynchronousRuleState:
+    """Stores pending points for asynchronous rules.
+    These are points which were requested but are not observed yet.
+    """
+
+    pending_points: TensorType
```

Comment:
Definitely mention that it can be `None` in the type. Whether to make it optional is up to you.

```suggestion
    pending_points: Optional[TensorType] = None
```

Reply:
Great catch, updated, as it can certainly be None